### PR TITLE
add error code from planner, expression and executor

### PR DESCRIPTION
**Pull request by ichn-hu**
### What problem does this PR solve? 

Associating internal error code for planner/core/errors.go, expression/errors.go and executor/errors.go to parser/mysql/errcode.go.

### What is changed and how it works?

Adding more error code and set up the mapping.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### CODE REVIEW

File: mysql/errname.go [hunk 1]
```diff
@@ -990,6 +990,20 @@ var MySQLErrName = map[uint16]string{
...
+	ErrUnsupportedType:             "Unsupported type %T",
+	ErrAnalyzeMissIndex:            "Index '%s' in field list does not exist in table '%s'",
+	ErrCartesianProductUnsupported: "Cartesian product is unsupported",
+	ErrPreparedStmtNotFound:        "Prepared statement not found",
+	ErrWrongParamCount:             "Wrong parameter count",
+	ErrSchemaChanged:               "Schema has changed",
+	ErrUnknownPlan:                 "Unknown plan",
+	ErrPrepareMulti:                "Can not prepare multiple statements",
+	ErrPrepareDDL:                  "Can not prepare DDL statements with parameters",
+	ErrResultIsEmpty:               "result is empty",
```
Comment (tangenta) [inline mysql/errname.go]:
```suggestion
	ErrResultIsEmpty:               "Result is empty",
```

File: mysql/errcode.go [hunk 2]
```diff
@@ -934,48 +934,61 @@ const (
...
+	ErrUnsupportedOp               = 8040
+	ErrRowNotFound                 = 8041
+	ErrTableStateCantNone          = 8042
+	ErrColumnStateNonPublic        = 8043
+	ErrIndexStateCantNone          = 8044
+	ErrInvalidRecordKey            = 8045
+	ErrColumnStateCantNone         = 8046
+	ErrUnsupportedValueForVar      = 8047
+	ErrUnsupportedIsolationLevel   = 8048
+	ErrUnsupportedType             = 8049
```
Comment (jackysp) [inline mysql/errcode.go]: It is better to start from 8100 to avoid some conflicts.

Comment (ichn-hu) [reply mysql/errcode.go]: OK, will do

Comment (ichn-hu) [reply mysql/errcode.go]: @jackysp but it seems starting from 8100 has already been taken

```
	ErrInvalidPluginID             = 8101
	ErrInvalidPluginManifest       = 8102
	ErrInvalidPluginName           = 8103
	ErrInvalidPluginVersion        = 8104
	ErrDuplicatePlugin             = 8105
	ErrInvalidPluginSysVarName     = 8106
	ErrRequireVersionCheckFail     = 8107
```

So do we want to start from 8108 or 8200?

**Comment by ichn-hu** [comment]
@tangenta @jackysp  PTAL again.

**Pull request merged**